### PR TITLE
Avoid returning secret value of the DevOps credential by creating credential mask

### LINKED ISSUE #299: Avoid return secret value of the DevOps credential
This is the backend part.

The API is `devops.kubesphere.io/v1alpha3/devops/test6r9qx/credentials/admin?content=1`

https://github.com/kubesphere/ks-devops/blob/be019b7a7c1b2917e9ec5216bc4143dd9551945e/pkg/kapis/devops/v1alpha3/register.go#L78

## References
* https://github.com/kubesphere/console/issues/2131

### PR DESCRIPTION
**Pull request by JohnNiang**
### What type of PR is this?

/kind bug
/area devops

### What this PR does / why we need it:

- Add credential mask;
- Apply the mask into listing, getting, updating and creating credential.

### Which issue(s) this PR fixes:

Fixes #299

#### Steps to test

Docker image for test:

```bash
johnniang/devops-apiserver:dev-v3.2.1-29e5c8b
```

1. Replace the docker image of `devops-apiserver` with test image above;
2. Create a credential for a DevOps Project and see response in browser debug network;
3. Update the credential and see response in browser debug network;
4. Back to credential list and see response in browser debug network;
5. Go into a credential and see response in browser debug network;

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
This feature is pretty important. I guess we can restart it. So, we can have more time to test it before the next release.

**Comment by JohnNiang** [comment]
> This feature is pretty important. I guess we can restart it. So, we can have more time to test it before the next release.

Agree. I will push it forward.

File: pkg/models/devops/devops.go [hunk 1]
```diff
@@ -292,15 +293,23 @@ func (d devopsOperator) CreateCredentialObj(projectName string, secret *v1.Secre
...
 	projectObj, err := d.ksclient.DevopsV1alpha3().DevOpsProjects().Get(d.context, projectName, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	return d.k8sclient.CoreV1().Secrets(projectObj.Status.AdminNamespace).Get(d.context, secretName, metav1.GetOptions{})
+	if secret, err := d.k8sclient.CoreV1().Secrets(projectObj.Status.AdminNamespace).Get(d.context, secretName, metav1.GetOptions{}); err != nil {
+		return nil, err
+	} else {
+		return secretutil.MaskCredential(secret), nil
+	}
```
Comment (LinuxSuRen) [inline pkg/models/devops/devops.go:312]: Currently, some APIs might rely on the secret content from this API. We can merge this PR if you can revert this part.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: pkg/models/devops/devops.go (hunk 1 above)
Comment (JohnNiang) [reply pkg/models/devops/devops.go:312]: Got it.

**Review by LinuxSuRen — COMMENTED** [review]
/approve

**Pull request merged**